### PR TITLE
Feature/connection wrapper

### PR DESCRIPTION
**Pull request by ilija-savic-devtech**
# What's the purpose of this MR?
Creating connection wrapper for MySQL and MongoDB Databasses
# Impact Analysis
(N/A)
# Where should the tester start?
- Run start_server.bat
- Go to bootstrap\bootstrap.php and change constant DATABASE_IN_USE to 'mysql' or 'mongodb'
- Run application with your browser on localhost
# How should this be manually tested?
Read above
# Any background context you want to provide?
(N/A)
# What are the relevant tickets?
(N/A)
# Screenshots (if appropriate)
(N/A)

### CODE REVIEW

**Comment by goran-erhartic-devtech** [comment]
Please use Manager class instead of Client class when instantiating mongodb, and remove "mongodb/mongodb": "^1" from composer.josn

After you make those changes, push them to this feature

**Comment by ilija-savic-devtech** [comment]
I pushed those changes, check it out

**Comment by goran-erhartic-devtech** [comment]
ok, so couple of things:
 - please create two separate script for a quick DB setup and population of table(s) for mysql and mongodb so I can test it out
 - refactor the code so you use singleton approach when instantiating the database
 - do not instantiante the database in index.php, that is why we have the bootstrap file, and while we are it, I would advise to create a separate config file for database and define constants there so we have a intuitive setup :)

**Comment by ilija-savic-devtech** [comment]
I used 'mongodump' command for exporting Mongo database. You can use 'mongorestore -d <database_name> <directory_backup>' command to import this database. Check out my DbConnection class, is this good approach?

**Comment by goran-erhartic-devtech** [comment]
I can see that you are still reworking your code, so once you are satisfied with it, ping me and I'll do the review

**Comment by ilija-savic-devtech** [comment]
Now i think it's ok. Take a look.

**Comment by goran-erhartic-devtech** [comment]
- In DbConnection class, you are passing Student instance as a parameter to CrudDatabaseMySql()... This means you are restricting your app to ONLY work with Student class - what would happen if you want to expand the app and perform CRUD actions on for example Professor?

- I can see you are not using templating engine anymore... Please keep using Twig or any other templating engine (except Blade)

- When declaring visibility, please keep 'public' functions at the top and 'protected'/'private' functions below them

**Comment by ilija-savic-devtech** [comment]
Something like this? :)

**Comment by goran-erhartic-devtech** [comment]
now you are just passing the student class in the connect() from the bootstrap file. the idea is to separate the two - BUT we need to inject the DB dependency when instantiating (in this case) the Student class.

**Comment by ilija-savic-devtech** [comment]
I think it's ok now, but i need your opinion :)

File: bootstrap/bootstrap.php [hunk 1]
```diff
@@ -1,2 +1,23 @@
...
+
+$loader = new Twig_Loader_Filesystem('..\html');
+$twig = new Twig_Environment($loader);
+
+$conn = \database\DbConnection::getInstance();
+$base = null;
+$crud = null;
+if (DATABASE_IN_USE == 'mysql') {
+	$base = $conn->connectMySql();
+	$crud = new \database\CrudDatabaseMySql($base, new \src\Student());
```
Comment (goran-erhartic-devtech) [inline bootstrap/bootstrap.php]: you keep insisting in passing Student instance to CrudDatabaseMySql counstructor, but you still keep instantiating the same object in that class... kinda defeats the purpose, don't you think? :) 
try renaming the "CrudDatabaseMySql" class to something like "StudentServiceMySql" - hopefully that hint will get you back to right track

File: bootstrap/bootstrap.php [hunk 2]
```diff
@@ -1,2 +1,23 @@
...
\ No newline at end of file
+require_once '..\vendor\autoload.php';
+require_once 'database_config.php';
+
+$loader = new Twig_Loader_Filesystem('..\html');
+$twig = new Twig_Environment($loader);
+
+$conn = \database\DbConnection::getInstance();
+$base = null;
+$crud = null;
```
Comment (goran-erhartic-devtech) [inline bootstrap/bootstrap.php]: ? :)

File: bootstrap/bootstrap.php [hunk 3]
```diff
@@ -1,2 +1,23 @@
...
-require_once '..\vendor\autoload.php';
\ No newline at end of file
+require_once '..\vendor\autoload.php';
+require_once 'database_config.php';
+
+$loader = new Twig_Loader_Filesystem('..\html');
+$twig = new Twig_Environment($loader);
+
+$conn = \database\DbConnection::getInstance();
+$base = null;
```
Comment (goran-erhartic-devtech) [inline bootstrap/bootstrap.php]: ? :)

**Comment by ilija-savic-devtech** [comment]
Is this ok? If it's ok, what's next :)

File: database/StudentServiceInterface.php [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+/**
+ * Created by PhpStorm.
+ * User: ilija.savic
+ * Date: 8/24/2017
+ * Time: 9:42 AM
+ */
+
+namespace database;
+
+interface StudentServiceInterface
```
Comment (goran-erhartic-devtech) [inline database/StudentServiceInterface.php:11]: is this interface only valid for Student model or? name of interface is kinda misleading

File: database/StudentServiceMySql.php [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+namespace database;
+
+use src\Student;
+
+class StudentServiceMySql implements StudentServiceInterface
+{
+	private $conn;
+
+	public function __construct($conn)
```
Comment (goran-erhartic-devtech) [inline database/StudentServiceMySql.php:11]: tip: you can use Typehinting when passing parameters -> let the __constructor() function know what type is $conn and you will gain access to it's methods

**Comment by goran-erhartic-devtech** [comment]
I will merge this feature - you can create new feature from develop branch and start working on getOne() and covering exceptions when trying to read from the database.

**Pull request merged**